### PR TITLE
tsp, make API internal, for multipart/form-data

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Treat it as internal API and no convenience API

### CODE REVIEW

File: typespec-tests/src/main/java/com/cadl/multipart/MultipartClient.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @return the {@link Response}.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<Void> uploadWithResponse(BinaryData request, RequestOptions requestOptions) {
+        // Protocol API does not work, as operation 'upload' is 'multipart/form-data'
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/main/java/com/cadl/multipart/MultipartClient.java:60]: Just say it won't work...

**Review by weidongxu-microsoft — COMMENTED** [review]
May replace the local test with that from cadl-ranch (when available).

File: typespec-tests/src/main/java/com/cadl/multipart/MultipartClient.java (hunk 1 above)
Comment (haolingdong-msft) [reply typespec-tests/src/main/java/com/cadl/multipart/MultipartClient.java:60]: If the protocol API does not work, why do we still generate it as `public`, probably make it package-private?

Comment (weidongxu-microsoft) [reply typespec-tests/src/main/java/com/cadl/multipart/MultipartClient.java:60]: make sense. change to internal as well.

**Review by haolingdong-msft — APPROVED** [review]

**Pull request merged**